src/Microsoft.Diagnostics.Tools.RuntimeClient/DiagnosticsIpc/IpcClient.cs

```diff
@@ -48,7 +49,8 @@ private static Stream GetTransport(int processId)
                     throw new PlatformNotSupportedException($"Process {processId} not running compatible .NET Core runtime");
                 }
                 string path = Path.Combine(Path.GetTempPath(), ipcPort);
-                var remoteEP = new UnixDomainSocketEndPoint(path);
+                Type unixDomainSocketEndPointType = typeof(Socket).Assembly.GetType("System.Net.Sockets.UnixDomainSocketEndPoint");
+                var remoteEP = (EndPoint)Activator.CreateInstance(unixDomainSocketEndPointType, new object[]{path});
```

Comment:
If the type is null then `ArgumentNullException` will be thrown. his would happen for Xamarin and Mono. This call is a netcoreapp specific thing. On Mono this call is `UnixEndPoint` for example. This would require a change to compile for them and play nicely. This change brings value in the sense that a NetFX tool could consume the runtime events or communicate through EventPipe with a 3+ runtime, but I am not sure that this is the right way to achieve that. Multi-TFM packages feel like a more appropriate solution if we don't plan to go away from the socket for unix. As for the transitive dependencies in library scenarios, that's fair and all, but I can't think of a good maintainable way to do this without reflection like here. Maybe someone with more experience in portable managed libraries has a better idea.

Reply:
Fixed the `ArgumentNullException` with an extra check.